### PR TITLE
Add synchronization script

### PR DESCRIPTION
**Pull request by i386x**
Add a script that propagates files that should be in sync across linux-system-roles repos. The propagating is done via pull request.

### CODE REVIEW

File: sync-template.sh [hunk 1]
```diff
@@ -0,0 +1,424 @@
+#!/bin/bash
+# SPDX-License-Identifier: MIT
+
+set -euo pipefail
+
+ME=$(basename $0)
+HERE=${PWD}
+WORKDIR="${HOME}/.cache/${ME%.*}"
+STDERR=/tmp/${ME%.*}.stderr
```
Comment (tyll) [inline sync-template.sh]: Files in `/tmp` need to be created with mktemp for security reasons, see for example: https://en.wikipedia.org/wiki/Symlink_race

**Comment by richm** [comment]
any updates?

**Comment by i386x** [comment]
@richm, the script is almost complete now. Things that should be done are:
- move `--clean` option out (provide it as a separate script instead)
- write a documentation (update `README.md`, provide some examples)

Things that should be done in future (not in this PR):
- provide scripts for repo health checking (like if there are no obsolete template files, conflicting branches, etc.)
- prevent copying of `lgtm.yml` if there are no Python files
- make a `tox` plugin that can help to reduce a large number of shell wrappers and the size of `tox.ini` itself

**Comment by i386x** [comment]
Also, a little cleanup in commit history (some commits, like license change, shoud be moved to separate pull request) before merging this will be useful.

**Comment by richm** [comment]
@i386x is this PR ready to merge?  Was there additional work required?

**Comment by i386x** [comment]
Hi Rich, I am planning to squash some commits to make a history more clear and also move some unrelated commits (like `Change license to MIT`) to separate pull requests.

**Comment by i386x** [comment]
I think this can be merged.

**Comment by richm** [comment]
> I think this can be merged.

Does this PR need to be rebased and squashed?  Or do we want all of those commits listed separately?

**Pull request merged**